pkg/api/mlflow/service/run/service.go

```diff
@@ -78,9 +78,19 @@ func (s Service) CreateRun(
 		return nil, api.NewBadRequestError("unable to parse experiment id '%s': %s", req.ExperimentID, err)
 	}
 
+	// experimentID 0 should be interpreted as the default for the namespace
+	if experimentID == 0 {
+		experimentID = int64(*ns.DefaultExperimentID)
+	}
+
 	experiment, err := s.experimentRepository.GetByNamespaceIDAndExperimentID(ctx, ns.ID, int32(experimentID))
 	if err != nil {
-		return nil, api.NewResourceDoesNotExistError("unable to find experiment with id '%s': %s", req.ExperimentID, err)
+		return nil, api.NewResourceDoesNotExistError(
+			"unable to find experiment '%d' for namespace '%s': %s",
+			experimentID,
+			ns.Code,
+			err,
```

Comment:
I would actually argue against changing this error message, unless we change this for _everything_